## 🩺 Diabetes Prediction using PySpark Machine Learning

With the rapid growth of data in healthcare, scalable machine learning solutions have become increasingly important. Traditional machine learning approaches may struggle with large datasets, making distributed computing frameworks like PySpark essential.

In this project, we build a diabetes prediction model using PySpark’s MLlib library on the Pima Indians Diabetes dataset. The dataset includes key medical features such as glucose levels, BMI, blood pressure, and age, which are critical indicators for diabetes diagnosis.

We utilize PySpark’s machine learning pipeline, including feature transformation with VectorAssembler and model training using Logistic Regression. The data is split into training and testing sets to evaluate model performance effectively.

🎯 **Goal:**  
To develop a scalable and efficient classification model for predicting diabetes using distributed machine learning techniques.

In [1]:
from pyspark.sql import SparkSession
from pyspark.ml.feature import VectorAssembler
from pyspark.ml.classification import LogisticRegression
from pyspark.ml.evaluation import BinaryClassificationEvaluator

In [3]:
spark = SparkSession.builder.appName("PimaIndianClassifier").getOrCreate()

data= spark.read.csv("/kaggle/input/datasets/nihalba/pima-indians-diabetes/pima-indians-diabetes.csv", inferSchema=True, header=True) #datayi okuttuk spark.read.csv
predictors = data.columns[:-1]
assembler = VectorAssembler(inputCols=predictors, outputCol="features") #vektore cevirdik
data = assembler.transform(data).select( "features", "Outcome")

train_data, test_data = data.randomSplit([0.7, 0.3], seed=42)
lr = LogisticRegression(labelCol="Outcome", featuresCol="features")
model = lr.fit(train_data)
predictions = model.transform(test_data)
evaluator = BinaryClassificationEvaluator(labelCol="Outcome")
accuracy = evaluator.evaluate(predictions)
print( "Accuracy: ", accuracy)

spark.stop()

Accuracy:  0.8546265328874024


##  Conclusion

In this project, we implemented a scalable machine learning pipeline using PySpark to predict diabetes outcomes. The Logistic Regression model achieved a strong performance with an evaluation score of approximately **0.85**, demonstrating the effectiveness of distributed machine learning even on moderately sized datasets.

Key takeaways:
- PySpark provides a powerful framework for building scalable machine learning pipelines  
- Logistic Regression performs well as a baseline model for binary classification problems  
- Feature transformation using VectorAssembler simplifies model input preparation  

💡 **Insights:**
- Medical features such as glucose levels and BMI play a significant role in prediction  
- Distributed processing ensures scalability for larger datasets  
